src/CoreFoundation/CFBundle.cs

```diff
@@ -608,5 +608,44 @@ static string FourCCToString (uint code)
 				}
 			}
 		}
+
+#if MONOMAC
+		[Introduced (PlatformName.MacOSX, 11, 0)]
+		[DllImport (Constants.CoreFoundationLibrary)]
+		[return: MarshalAs (UnmanagedType.I1)]
```

Comment:
What does this line do?

Reply:
@tj-devel709 Boolean types have different size depending in the language and platform, e.g.  the C `BOOL` on Win32 was 32bits, but it's often 1 byte, like in C++.

That can be a problem if the boolean value is written as 1 byte (e.g. on native side) then read as 4 bytes (e.g. on managed side). In this case the 3 other bytes (since it's often aligned on 4 bytes, even if it takes only one) content is undefined.

So, from the above example, `0-1-1-1` would be a `(bool)0x00` on native and `(bool)0x00010101` on managed which, beside being different, make things rather interesting at runtime :)

The `[MarshalAs]` attribute tells the runtime how to interpret the value, where `I1` means 1-byte integer.

Reply:
Thank you very much for the explanation!